src/python/WMCore/Storage/TrivialFileCatalog.py

```diff
@@ -196,55 +232,144 @@ def tfcFilename(contactString):
+                            result = proc['prefix']+'/$1'
+                            tfcInstance.addMapping(str(proc['protocol']), str(match), str(result), chain, 'lfn-to-pfn')
+                            #pfn-to-lfn
+                            match = proc['prefix']+'/(.*)'
+                            result = '/$1'
+                            tfcInstance.addMapping(str(proc['protocol']), str(match), str(result), chain, 'pfn-to-lfn')
+                        #here is rules  
+                        else:
+                            #loop over rules
+                            for rule in proc['rules']:
+                                match = rule['lfn']
+                                result = rule['pfn']
+                                tfcInstance.addMapping(str(proc['protocol']), str(match), str(result), chain, 'lfn-to-pfn')
+                                #pfn-to-lfn: not sure about this!!!
+                                match = rule['pfn'].replace('$1','(.*)')
+                                result = rule['lfn'].replace('/+','/').replace('^/','/')
+                                #now replace anything inside () with $1, for example (.*) --> $1, (store/.*) --> $1
+                                result = re.sub('\(.*\)','$1',result)
+                                tfcInstance.addMapping(str(proc['protocol']), str(match), str(result), chain, 'pfn-to-lfn')
+    
+    return tfcInstance
 
+def rseName(storageAttr):
+    rse = None
+    catalog = getCatalog(storageAttr)
+    storageJsonName = tfcFilename(catalog)
     try:
-        node = xmlFileToNode(filename)
+        jsonFile=open(storageJsonName,encoding="utf-8")
+        jsElements = json.load(jsonFile)
     except Exception as ex:
-        msg = "Error reading TrivialFileCatalog: %s\n" % filename
+        msg = "TrivialFileCatalog.py:rseName() Error reading FileCatalog: %s\n" % storageJsonName 
         msg += str(ex)
         raise RuntimeError(msg)
-
-    parsedResult = nodeReader(node)
-
-    tfcInstance = TrivialFileCatalog()
-    for mapping in ['lfn-to-pfn', 'pfn-to-lfn']:
-        for entry in parsedResult[mapping]:
-            protocol = entry.get("protocol", None)
-            match = entry.get("path-match", None)
-            result = entry.get("result", None)
-            chain = entry.get("chain", None)
-            if True in (protocol, match == None):
-                continue
-            tfcInstance.addMapping(str(protocol), str(match), str(result), chain, mapping)
-    return tfcInstance
-
-
-def loadTFC(contactString):
+    for jsElement in jsElements:
+        if jsElement['site'] == storageAttr['storageSite'] and jsElement['volume'] == storageAttr['volume']:
+            rse = jsElement['rse']
+            break
+    return rse
+
+def lfnPrefix(storageAttr):
+    catalog = getCatalog(storageAttr)
+    #now get lfn prefix
+    storageJsonName = tfcFilename(catalog)
+    try:
+        jsonFile = open(storageJsonName,encoding="utf-8")
```

Comment:
Same comment as the one above, please replace it by a context manager.

Reply:
Yes, I will